packages/lib/src/components/internal/UIElement/UIElement.tsx

```diff
@@ -83,7 +83,11 @@ export abstract class UIElement<P extends UIElementProps = UIElementProps> exten
             ...componentProps
         };
 
-        this.props = this.formatProps({ ...this.constructor['defaultProps'], ...finalProps });
+        this.props = this.formatProps({
+            ...this.constructor['defaultProps'], // component defaults
+            ...this.core.getRegulatoryDefaults(), // regulatory defaults
```

Comment:
Does the core need to know about this `getRegulatoryDefaults` function? 
We could skip adding an extra public function in the core. We could pull this directly from here: 

```js
this.props = this.formatProps({
  ...this.constructor['defaultProps'], // component defaults
  ...this.getRegulatoryDefaults({ countryCode: this.core.options.contryCode, isDropin: this.type === 'dropin'  }), 
  ...finalProps
});
```

Reply:
I had it in a separate, standalone function before, but then thought since we do things like `this.core.getPropsForComponent`  - isn't this retrieving and compiling the correct props for any component part of the core's responsibility?

Reply:
I'd say that the UIElement is responsible to build the properties of the UI component and retrieve from where it is necessary.

On core we have the `getCorePropsForComponent`, which basically returns the properties that belongs only to the core module: the initialized modules; the properly parsed loadingContext, cdnContext; createFromAction method, etc.

From my view, in this case, it is fine to just pull from an standalone function directly into the UIElement. Core does not need to know about these regulatory props. Another benefit is that we do not have to expose another public function of the core, as exposing public functions can lead to typescript merchants trying to use it for something else.

Reply:
Made changes after our discussion